### Assessing Lecturer Competency in Hybrid Learning Environments Using Decision Support System Approach Based on Student Feedback Questionnaire

### Mengevaluasi Kompetensi Dosen dalam Lingkungan Pembelajaran Hybrid Menggunakan Pendekatan Decision Support System Berdasarkan Kuesioner Umpan Balik Mahasiswa

# Dataset Analysis

In [1]:
# import csv
import pandas as pd
import numpy as np
import pingouin as pg

# read xls file
df = pd.read_excel('quiz_20221.xlsx')
df

dosen     ta             nim       kdmk  \
0      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  20221  A11.2017.10120   AF201703   
1      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  20221  A11.2017.10350   AF201703   
2      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  20221  A11.2018.11309   AF201703   
3      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  20221  A11.2019.11618   AF201703   
4      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  20221  A11.2019.11622   AF201703   
...                                     ...    ...             ...        ...   
20092         DIBYO ADI WIBOWO, S.Si., M.Si  20221  A11.2020.80018  A11.54508   
20093         DIBYO ADI WIBOWO, S.Si., M.Si  20221  A11.2020.80018  A11.54812   
20094         DEWI PERGIWATI, S.Kom., M.Kom  20221  A11.2018.11461   AF201704   
20095         DEWI PERGIWATI, S.Kom., M.Kom  20221  A11.2021.13607   AF201704   
20096              AYU ASHARI, S.S.T, M.Kes  20221  A11.2020.13081    U201701   

                        nmmk  q1  q2  q3  q4  q5  q6  q7  q8  q9  q10  \
0          TECHNOPRENEURSHIP   2   2   2   2   2   2   2   2   2    2   
1          TECHNOPRENEURSHIP   4   4   4   4   4   4   4   4   4    4   
2          TECHNOPRENEURSHIP   4   4   4   4   4   4   4   4   4    4   
3          TECHNOPRENEURSHIP   4   4   4   4   4   4   4   4   4    4   
4          TECHNOPRENEURSHIP   4   4   4   4   4   4   4   4   4    4   
...                      ...  ..  ..  ..  ..  ..  ..  ..  ..  ..  ...   
20092     STRATEGI ALGORITMA   4   3   3   4   4   4   4   4   4    4   
20093         METODE NUMERIK   4   4   4   4   4   4   4   4   4    4   
20094  DASAR DASAR KOMPUTASI   4   4   4   4   4   4   4   4   4    4   
20095  DASAR DASAR KOMPUTASI   4   4   4   4   4   4   3   3   3    4   
20096    DASAR KEWIRAUSAHAAN   3   3   3   4   3   3   3   3   3    3   

           saran  
0            NaN  
1            NaN  
2            NaN  
3             ok  
4            NaN  
...          ...  
20092  Tidak ada  
20093  Tidak ada  
20094        NaN  
20095        NaN  
20096        NaN  

[20097 rows x 16 columns]

In [2]:
df2 = df.copy()
df2 = df2.drop(['ta','nmmk','saran'], axis=1)
df2

dosen             nim       kdmk  q1  \
0      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  A11.2017.10120   AF201703   2   
1      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  A11.2017.10350   AF201703   4   
2      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  A11.2018.11309   AF201703   4   
3      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  A11.2019.11618   AF201703   4   
4      Prof. Dr. Ir EDI NOERSASONGKO, M.Kom  A11.2019.11622   AF201703   4   
...                                     ...             ...        ...  ..   
20092         DIBYO ADI WIBOWO, S.Si., M.Si  A11.2020.80018  A11.54508   4   
20093         DIBYO ADI WIBOWO, S.Si., M.Si  A11.2020.80018  A11.54812   4   
20094         DEWI PERGIWATI, S.Kom., M.Kom  A11.2018.11461   AF201704   4   
20095         DEWI PERGIWATI, S.Kom., M.Kom  A11.2021.13607   AF201704   4   
20096              AYU ASHARI, S.S.T, M.Kes  A11.2020.13081    U201701   3   

       q2  q3  q4  q5  q6  q7  q8  q9  q10  
0       2   2   2   2   2   2   2   2    2  
1       4   4   4   4   4   4   4   4    4  
2       4   4   4   4   4   4   4   4    4  
3       4   4   4   4   4   4   4   4    4  
4       4   4   4   4   4   4   4   4    4  
...    ..  ..  ..  ..  ..  ..  ..  ..  ...  
20092   3   3   4   4   4   4   4   4    4  
20093   4   4   4   4   4   4   4   4    4  
20094   4   4   4   4   4   4   4   4    4  
20095   4   4   4   4   4   3   3   3    4  
20096   3   3   4   3   3   3   3   3    3  

[20097 rows x 13 columns]

## Check Unique Classes Count

In [3]:
df3 = df.copy()
dosen = df3['dosen'].unique()
students = df3['nim'].unique()
classes = df3['kdmk'].unique()

print("unique classes", len(classes))
print("student count", len(students))
print("lecturer count", len(dosen))

unique classes 42
student count 3143
lecturer count 135


In [22]:
df4 = df.copy()

for i in range (len(df4)):
    df4.loc[i, 'dosen'] = df4.loc[i, 'dosen'] + str(" - ") + df4.loc[i, 'kdmk']

print(len(df4['dosen'].unique()))

['Prof. Dr. Ir EDI NOERSASONGKO, M.Kom - AF201703'
 'SUDARYANTO, M.Kom - AF201702' 'Dr.  ABDUL SYUKUR - A11.54205'
 'EDY MULYANTO, SSi, M.Kom - A11.54302'
 'ZAENAL ARIFIN, SE, M.Kom - AF201703'
 'ZAENAL ARIFIN, SE, M.Kom - AF201702'
 'Dr. YUVENTIUS TYAS CATUR PRAMUDI, S.Si., M.Kom - A11.54506'
 'ERLIN DOLPHINA, SH, MM - A11.54205'
 'PURWANTO, S.Si, M.Kom, Ph.D - A11.54303'
 'PURWANTO, S.Si, M.Kom, Ph.D - A11.54812'
 'PURWANTO, S.Si, M.Kom, Ph.D - A11.54302'
 'MOHAMAD SIDIQ, S.Si, M.Kom - A11.54506'
 'Dr. ARIS MARJUNI, S.Si., M.Kom - A11.54508'
 'Dr. ARIS MARJUNI, S.Si., M.Kom - A11.54303'
 'EDI FAISAL, S.Kom, M.Kom - A11.54403'
 'EDI FAISAL, S.Kom, M.Kom - AF201704'
 'ACHMAD WAHID KURNIAWAN, S.Si, M.Kom - A11.54101'
 'Dr. Ir. ARIPIN, M.Kom - A11.54505' 'AYU PERTIWI, S.Kom, MT - A11.54828'
 'AYU PERTIWI, S.Kom, MT - A11.54504'
 'DR. FIKRI BUDIMAN, M.Kom - A11.54506'
 'Dr M. ARIEF SOELEMAN, M.Kom - A11.54505'
 'Dr M. ARIEF SOELEMAN, M.Kom - A11.54403' 'MUSLIH, M.Kom - AF201704'
 'Dr. NOV

### Check Overall Cronbach Alpha Based on Selected Aspect  

The two aspects that we selected are as follows:  
- Knowledge Mastery, how well the lecturer knows about the academic material. Covered by P1,P2,P3,P5,P10
- Teaching Skill, how well the lecturer transfers the knowledge to the students. Covered by P4,P6,P7,P8,P9 

In [5]:
df_km = df3.copy()
df_km = df_km.drop(['q4','q5','q6','q7','q8','q9','dosen','ta','nim','kdmk','nmmk','saran'], axis=1)
df_km

q1  q2  q3  q10
0       2   2   2    2
1       4   4   4    4
2       4   4   4    4
3       4   4   4    4
4       4   4   4    4
...    ..  ..  ..  ...
20092   4   3   3    4
20093   4   4   4    4
20094   4   4   4    4
20095   4   4   4    4
20096   3   3   3    3

[20097 rows x 4 columns]

In [6]:
pg.cronbach_alpha(data=df_km)

(0.9059330153141515, array([0.904, 0.908]))

In [7]:
df_ts = df3.copy()
df_ts = df_ts.drop(['q1','q2','q3','q5','q10','dosen','ta','nim','kdmk','nmmk','saran'],axis=1)
df_ts

q4  q6  q7  q8  q9
0       2   2   2   2   2
1       4   4   4   4   4
2       4   4   4   4   4
3       4   4   4   4   4
4       4   4   4   4   4
...    ..  ..  ..  ..  ..
20092   4   4   4   4   4
20093   4   4   4   4   4
20094   4   4   4   4   4
20095   4   4   3   3   3
20096   4   3   3   3   3

[20097 rows x 5 columns]

In [8]:
pg.cronbach_alpha(data=df_ts)

(0.8954850845946283, array([0.893, 0.898]))

In [9]:
import itertools

df3.drop(['dosen','ta','nim','kdmk','nmmk','saran'], axis=1, inplace=True)

# Create an array of 10 numbers
numbers = ['q1', 'q2', 'q3', 'q4', 'q5', 'q6', 'q7', 'q8', 'q9', 'q10']

df_ca_test = pd.DataFrame(columns=['combination1', 'CA1score', 'combination2', 'CA2score', 'totalCA'])

# Get all possible permutations of the array
combinations = itertools.combinations(numbers,5)

# Loop through all permutations and print them
for combination1 in combinations:

    #copy original dataframe
    df_ca = df3.copy()

    #extract items from tuple
    combination1 = list(combination1)

    #eliminate combination list from numbers list
    combination2 = numbers.copy()
    for item in combination1:
        combination2.remove(item)

    # #drop columns
    df_ca1 = df_ca.drop(combination1, axis=1)
    df_ca2 = df_ca.drop(combination2, axis=1)

    #calculate cronbach alpha
    ca_score1 = pg.cronbach_alpha(data=df_ca1)
    ca_score2 = pg.cronbach_alpha(data=df_ca2)

    # round to 2 decimal places
    ca_score1 = round(ca_score1[0], 2)
    ca_score2 = round(ca_score2[0], 2)

    # calculate sum of ca_score
    total_ca = ca_score1 + ca_score2

    #add all data to dataframe
    df_ca_test.loc[len(df_ca_test)] = [combination1, ca_score1, combination2, ca_score2, total_ca]
    

df_ca_test

combination1  CA1score           combination2  CA2score  totalCA
0     [q1, q2, q3, q4, q5]      0.92  [q6, q7, q8, q9, q10]      0.90     1.82
1     [q1, q2, q3, q4, q6]      0.93  [q5, q7, q8, q9, q10]      0.89     1.82
2     [q1, q2, q3, q4, q7]      0.92  [q5, q6, q8, q9, q10]      0.90     1.82
3     [q1, q2, q3, q4, q8]      0.92  [q5, q6, q7, q9, q10]      0.90     1.82
4     [q1, q2, q3, q4, q9]      0.91  [q5, q6, q7, q8, q10]      0.90     1.81
..                     ...       ...                    ...       ...      ...
247  [q5, q6, q7, q8, q10]      0.90   [q1, q2, q3, q4, q9]      0.91     1.81
248  [q5, q6, q7, q9, q10]      0.90   [q1, q2, q3, q4, q8]      0.92     1.82
249  [q5, q6, q8, q9, q10]      0.90   [q1, q2, q3, q4, q7]      0.92     1.82
250  [q5, q7, q8, q9, q10]      0.89   [q1, q2, q3, q4, q6]      0.93     1.82
251  [q6, q7, q8, q9, q10]      0.90   [q1, q2, q3, q4, q5]      0.92     1.82

[252 rows x 5 columns]

We obtained all the possible combinations of the two-split of 10 questions in our questionnaire.  
Result shows that there are only 3 variance of the total score (from 2 aspects), and the combinations we chose earlier has the highest total score of 1.82.

The cronbach alpha score for the two aspects of our questionnaire is 0.91 and 0.89. This means the equivalence of each construct in the questionnaire is very high.

In [10]:
result = pd.DataFrame(columns=['construct', 'CA score'])

# add some daata to result daataframe
result.loc[len(result)] = ['Knowledge Mastery', 0.91]
result.loc[len(result)] = ['Teaching Skill', 0.89]

result

construct  CA score
0  Knowledge Mastery      0.91
1     Teaching Skill      0.89

Assessing Lecturer Competency in Hybrid Learning Environments Using Decision Support System Approach Based on Student Feedback Questionnaire